hw/xbox/nv2a/pgraph.c

```diff
@@ -5554,6 +5555,30 @@ static void surface_copy_expand(uint8_t *out, uint8_t *in, unsigned int width,
     }
 }
 
+void export_pgraph_data(NV2AState *d, SurfaceBinding *surface)
+{
+    //FIXME: correct memory allocation for imagedata.
+    //FIXME: Export correctly data from pg and Surface into ExportData
+    PGRAPHState *pg = &d->pgraph;
+    ExportData *imagedata = (ExportData*)malloc(pg->surface_shape.clip_x * pg->surface_shape.clip_y * pg->surface_shape.clip_width * pg->surface_shape.clip_height 
+                            * surface->fmt.bytes_per_pixel * pg->surface_scale_factor * surface->fmt.gl_format * surface->fmt.gl_type *
+                            surface->vram_addr * surface->width *  surface->height);
+
+    imagedata->clip_x = pg->surface_shape.clip_x;
+    imagedata->clip_y = pg->surface_shape.clip_y;
+    imagedata->clip_width = pg->surface_shape.clip_width;
+    imagedata->clip_width = pg->surface_shape.clip_height;
+    imagedata->bytes_per_pixel = surface->fmt.bytes_per_pixel;
+    imagedata->surface_scale_factor = pg->surface_scale_factor;
+    imagedata->gl_format = surface->fmt.gl_format;
+    imagedata->gl_type = surface->fmt.gl_type;
+    imagedata->vram_addr = surface->vram_addr;
+    imagedata->width = surface->width;
+    imagedata->height = surface->height;
+
+    free(imagedata);
```

Comment:
Not sure what the point of allocating `imagedata` is if you're free-ing it in the same function. What are you trying to do with that struct?